lib/install.js

```diff
@@ -834,7 +834,7 @@ Installer.prototype.printInstalledForHuman = function (diffs, auditResult) {
   if (removed) actions.push('removed ' + packages(removed))
   if (updated) actions.push('updated ' + packages(updated))
   if (moved) actions.push('moved ' + packages(moved))
-  if (auditResult && auditResult.metadata.totalDependencies) {
+  if (auditResult && auditResult.metadata && auditResult.metadata.totalDependencies) {
```

Comment:
If this is missing, there's definitely an issue with the endpoint. I'd rather spit out a warning here if this happens, so we can make sure to track it down. In the future, it might also be good to have [`npm-audit-report`](https://github.com/npm/cli/pull/1) verify the data we get from the server and warn accordingly, and we can remove the warning from `lib/install.js` at that point.

Reply:
There could also be a bug in a non-npm registry - like artifactory - if they implemented audit support but did it incorrectly?

Reply:
yes, the intention is also to cover non-npm registries, but I didn't think this was necessarily an artifactory issue. If so, then yeah, they should fix their stuff.